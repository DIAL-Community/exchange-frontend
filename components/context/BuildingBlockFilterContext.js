import React, { createContext, useState } from 'react'

const BuildingBlockFilterContext = createContext()
const BuildingBlockFilterDispatchContext = createContext()

const BuildingBlockFilterProvider = ({ children }) => {
  const [showMature, setShowMature] = useState(false)
  const [sdgs, setSDGs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [categoryTypes, setCategoryTypes] = useState([])

  const [search, setSearch] = useState('')
  const [sortColumn, setSortColumn] = useState('name')
  const [sortDirection, setSortDirection] = useState('asc')

  const setSdgs = setSDGs

  const buildingBlockFilterValues = {
    showMature, sdgs, useCases, workflows, categoryTypes, search, sortColumn, sortDirection
  }
  const buildingBlockFilterDispatchValues = {
    setShowMature,
    setSDGs,
    setSdgs,
    setUseCases,
    setWorkflows,
    setCategoryTypes,
    setSearch,
    setSortColumn,
    setSortDirection
  }

  return (
    <BuildingBlockFilterContext.Provider value={buildingBlockFilterValues}>
      <BuildingBlockFilterDispatchContext.Provider value={buildingBlockFilterDispatchValues}>
        {children}
      </BuildingBlockFilterDispatchContext.Provider>
    </BuildingBlockFilterContext.Provider>
  )
}

export { BuildingBlockFilterProvider, BuildingBlockFilterContext, BuildingBlockFilterDispatchContext }
