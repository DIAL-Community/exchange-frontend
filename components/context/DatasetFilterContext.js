import { createContext, useState } from 'react'

const DatasetFilterContext = createContext()
const DatasetFilterDispatchContext = createContext()

const DatasetFilterProvider = ({ children }) => {
  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [tags, setTags] = useState([])
  const [datasetTypes, setDatasetTypes] = useState([])

  const [search, setSearch] = useState('')

  const [pageNumber, setPageNumber] = useState(0)
  const [pageOffset, setPageOffset] = useState(0)

  const datasetFilterValues = {
    origins,
    countries,
    sectors,
    organizations,
    sdgs,
    tags,
    datasetTypes,
    search,
    pageOffset,
    pageNumber
  }
  const datasetFilterDispatchValues = {
    setOrigins,
    setCountries,
    setSectors,
    setOrganizations,
    setSdgs,
    setTags,
    setDatasetTypes,
    setSearch,
    setPageNumber,
    setPageOffset
  }

  return (
    <DatasetFilterContext.Provider value={datasetFilterValues}>
      <DatasetFilterDispatchContext.Provider value={datasetFilterDispatchValues}>
        {children}
      </DatasetFilterDispatchContext.Provider>
    </DatasetFilterContext.Provider>
  )
}

export { DatasetFilterProvider, DatasetFilterContext, DatasetFilterDispatchContext }
