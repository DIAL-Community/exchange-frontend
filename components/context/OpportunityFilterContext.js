import React, { createContext, useState } from 'react'

const OpportunityFilterContext = createContext()
const OpportunityFilterDispatchContext = createContext()

const OpportunityFilterProvider = ({ children }) => {
  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [countries, setCountries] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sectors, setSectors] = useState([])
  const [tags, setTags] = useState([])
  const [useCases, setUseCases] = useState([])
  const [showClosed, setShowClosed] = useState(false)

  const [search, setSearch] = useState('')
  const [sortColumn, setSortColumn] = useState('name')
  const [sortDirection, setSortDirection] = useState('asc')

  const opportunityFilterValues = {
    buildingBlocks,
    countries,
    organizations,
    sectors,
    tags,
    useCases,
    showClosed,
    search,
    sortColumn,
    sortDirection
  }
  const opportunityFilterDispatchValues = {
    setBuildingBlocks,
    setCountries,
    setOrganizations,
    setSectors,
    setTags,
    setUseCases,
    setShowClosed,
    setSearch,
    setSortColumn,
    setSortDirection
  }

  return (
    <OpportunityFilterContext.Provider value={opportunityFilterValues}>
      <OpportunityFilterDispatchContext.Provider value={opportunityFilterDispatchValues}>
        {children}
      </OpportunityFilterDispatchContext.Provider>
    </OpportunityFilterContext.Provider>
  )
}

export { OpportunityFilterProvider, OpportunityFilterContext, OpportunityFilterDispatchContext }
