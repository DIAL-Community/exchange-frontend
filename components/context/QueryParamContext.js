import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
