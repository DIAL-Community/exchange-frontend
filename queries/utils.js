const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
