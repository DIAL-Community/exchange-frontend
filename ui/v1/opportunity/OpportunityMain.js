import OpportunityMainLeft from './OpportunityMainLeft'
import OpportunityMainRight from './OpportunityMainRight'

const OpportunityMain = ({ activeTab }) => {
  return (
    <div className='px-4 lg:px-8 xl:px-56'>
      <div className='grid grid-cols-3 gap-x-8'>
        <div className='hidden md:block col-span-1'>
          <OpportunityMainLeft activeTab={activeTab} />
        </div>
        <div className='col-span-3 md:col-span-2'>
          <OpportunityMainRight activeTab={activeTab} />
        </div>
      </div>
    </div>
  )
}

export default OpportunityMain
