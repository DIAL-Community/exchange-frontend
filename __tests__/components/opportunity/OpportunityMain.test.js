import { act } from 'react-dom/test-utils'
import { screen } from '@testing-library/dom'
import { render } from '../../test-utils'
import { mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import OpportunityMain from '../../../components/opportunity/OpportunityMain'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import {
  OPPORTUNITY_PAGINATION_ATTRIBUTES_QUERY,
  PAGINATED_OPPORTUNITIES_QUERY
} from '../../../components/shared/query/opportunity'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { OpportunityFilterProvider } from '../../../components/context/OpportunityFilterContext'
import { paginatedOpportunities, opportunityPaginationAttribute } from './data/OpportunityMain.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the opportunity main page.', () => {
  test('Should render list of opportunities.', async () => {
    const mockOpportunityPaginationAttribute = generateMockApolloData(
      OPPORTUNITY_PAGINATION_ATTRIBUTES_QUERY,
      {
        search: '',
        countries: [],
        buildingBlocks: [],
        organizations: [],
        useCases: [],
        sectors: [],
        tags: [],
        showClosed: false,
        showGovStackOnly: false
      },
      null,
      opportunityPaginationAttribute
    )
    const mockPaginatedOpportunities = generateMockApolloData(
      PAGINATED_OPPORTUNITIES_QUERY,
      {
        search: '',
        countries: [],
        buildingBlocks: [],
        organizations: [],
        useCases: [],
        sectors: [],
        tags: [],
        showClosed: false,
        showGovStackOnly: false,
        limit: 8,
        offset: 0
      },
      null,
      paginatedOpportunities
    )

    const { container } = render(
      <CustomMockedProvider mocks={[mockPaginatedOpportunities, mockOpportunityPaginationAttribute]}>
        <QueryParamContextProvider>
          <OpportunityFilterProvider>
            <OpportunityMain activeTab={0} />
          </OpportunityFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    await act(() => new Promise((resolve) => setTimeout(resolve, 30)))

    expect(await screen.findByText('Make your supply chain more sustainable')).toBeInTheDocument()
    expect(await screen.findByText('Improving cross-border e-Commerce services')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
