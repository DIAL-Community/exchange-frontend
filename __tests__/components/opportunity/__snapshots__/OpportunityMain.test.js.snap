// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the opportunity main page. Should render list of opportunities. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="px-4 lg:px-8 xl:px-56"
  >
    <div
      class="grid grid-cols-3 gap-x-8"
    >
      <div
        class="hidden md:block col-span-1"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  Filter your results by
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Use Case
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Building Block
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Organization
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Sector
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  <button
                    type="button"
                  >
                    <div
                      class="flex gap-3 text-dial-sapphire"
                    >
                      <div
                        class="my-auto"
                      >
                        Refine with more filters
                      </div>
                      <div
                        class="ml-auto py-2 text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M201.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 274.7 86.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Bookmark
              </div>
              <button
                class="group flex flex-row gap-x-3"
                type="button"
              >
                <img
                  alt="Logo of Bookmark."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/bookmark-icon.svg"
                  width="40"
                />
                <div
                  class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                >
                  <div
                    class="text-sm text-dial-stratos text-left"
                  >
                    Bookmark this page.
                  </div>
                </div>
              </button>
              <div
                class="text-sm text-dial-stratos"
              >
                Your bookmarks can be found in your Account page.
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Share
              </div>
              <div
                class="flex flex-row gap-3"
              >
                <button>
                  <img
                    alt="Logo of LinkedIn."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/linkedin-icon.svg"
                    width="40"
                  />
                </button>
                <button>
                  <div
                    class="bg-dial-iris-blue rounded-full w-[40px] h-[40px]"
                  >
                    <div
                      class="flex items-center h-full"
                    >
                      <img
                        alt="Logo of WhatsApp."
                        class="my-auto mx-auto"
                        height="20"
                        src="/ui/v1/whatsapp-icon.svg"
                        width="20"
                      />
                    </div>
                  </div>
                </button>
                <button>
                  <img
                    alt="Logo of Email."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/email-icon.svg"
                    width="40"
                  />
                </button>
                <button>
                  <img
                    alt="Logo of Copy."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/copy-icon.svg"
                    width="40"
                  />
                </button>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-3 md:col-span-2"
      >
        <div
          class="min-h-[50vh]"
        >
          <div
            class="py-3"
          >
            <div
              class="flex flex-row gap-3"
            >
              <div
                class="block md:hidden relative"
              >
                <button
                  class="my-auto h-full"
                  type="button"
                >
                  <svg
                    class="text-2xl text-dial-plum mx-auto"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 416c0 17.7 14.3 32 32 32l54.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48L480 448c17.7 0 32-14.3 32-32s-14.3-32-32-32l-246.7 0c-12.3-28.3-40.5-48-73.3-48s-61 19.7-73.3 48L32 384c-17.7 0-32 14.3-32 32zm128 0a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zM320 256a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zm32-80c-32.8 0-61 19.7-73.3 48L32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l246.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48l54.7 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-54.7 0c-12.3-28.3-40.5-48-73.3-48zM192 128a32 32 0 1 1 0-64 32 32 0 1 1 0 64zm73.3-64C253 35.7 224.8 16 192 16s-61 19.7-73.3 48L32 64C14.3 64 0 78.3 0 96s14.3 32 32 32l86.7 0c12.3 28.3 40.5 48 73.3 48s61-19.7 73.3-48L480 128c17.7 0 32-14.3 32-32s-14.3-32-32-32L265.3 64z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="ml-auto w-full"
              >
                <label
                  class="relative"
                  for="search-bar"
                >
                  <input
                    class="w-full border text-sm"
                    id="search-bar"
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                  <svg
                    class="absolute top-1/2 transform -translate-y-1/2 right-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </label>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div>
              <div
                class="relative"
              >
                <a
                  href="/opportunities/call_for_digital_products_to_imp"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] bg-dial-violet"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-dial-plum rounded-full"
                      >
                        <img
                          alt="Logo of RFP."
                          class="object-contain w-8 h-8 mx-auto mt-6 white-filter"
                          src="/ui/v1/opportunity-header.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-plum"
                        >
                          Call for Digital Products in Somalia
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex flex-row gap-x-3"
                        >
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              RFP Status
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              OPEN
                            </div>
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              Closing Date
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              10/4/2023
                            </div>
                          </div>
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            Sectors
                             (
                            1
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Organizations
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            1
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-dial-iris-blue"
                        >
                          <div
                            class="border border-dial-iris-blue text-dial-iris-blue rounded px-6 py-1"
                          >
                            GOVSTACK
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/opportunities/greening_and_advancing_ecommerce"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] false"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-dial-plum rounded-full"
                      >
                        <img
                          alt="Logo of RFP."
                          class="object-contain w-8 h-8 mx-auto mt-6 white-filter"
                          src="/ui/v1/opportunity-header.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-plum"
                        >
                          Greening and advancing e-commerce in Ghana
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex flex-row gap-x-3"
                        >
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              RFP Status
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              OPEN
                            </div>
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              Closing Date
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              10/6/2023
                            </div>
                          </div>
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            Sectors
                             (
                            2
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Organizations
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            4
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-dial-iris-blue"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/opportunities/improving_crossborder_ecommerce"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] bg-dial-violet"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-dial-plum rounded-full"
                      >
                        <img
                          alt="Logo of RFP."
                          class="object-contain w-8 h-8 mx-auto mt-6 white-filter"
                          src="/ui/v1/opportunity-header.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-plum"
                        >
                          Improving cross-border e-Commerce services
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex flex-row gap-x-3"
                        >
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              RFP Status
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              OPEN
                            </div>
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              Closing Date
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              10/7/2023
                            </div>
                          </div>
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            Sectors
                             (
                            2
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Organizations
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            3
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-dial-iris-blue"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/opportunities/make_your_supply_chain_more_sust"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] false"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-dial-plum rounded-full"
                      >
                        <img
                          alt="Logo of RFP."
                          class="object-contain w-8 h-8 mx-auto mt-6 white-filter"
                          src="/ui/v1/opportunity-header.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-plum"
                        >
                          Make your supply chain more sustainable
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex flex-row gap-x-3"
                        >
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              RFP Status
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              OPEN
                            </div>
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              Closing Date
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              10/24/2023
                            </div>
                          </div>
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            Sectors
                             (
                            4
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Organizations
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            9
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-dial-iris-blue"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/opportunities/call_for_contributors_to_govstack"
                >
                  <div
                    class="px-4 py-6 rounded-lg min-h-[13.5rem] bg-dial-violet"
                  >
                    <div
                      class="flex flex-col lg:flex-row gap-x-6 gap-y-3"
                    >
                      <div
                        class="w-20 h-20 mx-auto bg-dial-plum rounded-full"
                      >
                        <img
                          alt="Logo of RFP."
                          class="object-contain w-8 h-8 mx-auto mt-6 white-filter"
                          src="/ui/v1/opportunity-header.svg"
                        />
                      </div>
                      <div
                        class="flex flex-col gap-y-3 max-w-3xl lg:w-10/12"
                      >
                        <div
                          class="text-lg font-semibold text-dial-plum"
                        >
                          Call for Contributors to Consent Building Block
                        </div>
                        <div
                          class="line-clamp-4 text-dial-stratos"
                        >
                          Some parsed description.
                        </div>
                        <div
                          class="flex flex-row gap-x-3"
                        >
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              RFP Status
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              OPEN
                            </div>
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="flex flex-row gap-x-2"
                          >
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              Closing Date
                              :
                            </div>
                            <div
                              class="text-sm text-dial-stratos"
                            >
                              10/14/2023
                            </div>
                          </div>
                        </div>
                        <div
                          class="flex gap-x-2 text-dial-stratos"
                        >
                          <div
                            class="text-sm"
                          >
                            Sectors
                             (
                            1
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Organizations
                             (
                            0
                            )
                          </div>
                          <div
                            class="border-r border-dial-slate-400"
                          />
                          <div
                            class="text-sm"
                          >
                            Countries
                             (
                            1
                            )
                          </div>
                        </div>
                        <div
                          class="flex gap-2 text-xs text-dial-iris-blue"
                        >
                          <div
                            class="border border-dial-iris-blue text-dial-iris-blue rounded px-6 py-1"
                          >
                            GOVSTACK
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col xl:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm font-semibold"
              >
                Showing 1 - 8 of 14 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm text-dial-slate-300 font-semibold"
                role="navigation"
              >
                <li
                  class="border border-current rounded-md opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500 "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="border border-current rounded-md text-dial-iris-blue"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 2"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    2
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-disabled="false"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
